lib/time/time.go

```diff
@@ -0,0 +1,462 @@
+		"millisecond": durInt(time.Millisecond),
+		"second":      durInt(time.Second),
+		"minute":      durInt(time.Minute),
+		"hour":        durInt(time.Hour),
+	},
+}
+
+func durInt(d time.Duration) starlark.Int {
+	return starlark.MakeInt64(int64(d))
+}
+
+// LoadModule loads the time module.
+// It is concurrency-safe and idempotent.
+func LoadModule() (starlark.StringDict, error) {
+	return starlark.StringDict{
+		ModuleName: Module,
+	}, nil
+}
+
+// NowFunc is a function that generates the current time. Intentionally exported
+// so that it can be overridden, for example by applications that require their
+// Starlark scripts to be fully deterministic.
+var NowFunc = time.Now
+
+func parseDuration(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var s starlark.String
+	if err := starlark.UnpackArgs("duration", args, kwargs, "s", &s); err != nil {
+		return nil, err
+	}
+
+	d, err := time.ParseDuration(string(s))
+	if err != nil {
+		return nil, err
+	}
+
+	return Duration(d), nil
+}
+
+func parseLocation(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.String
+	if err := starlark.UnpackArgs("location", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+	loc, err := time.LoadLocation(string(x))
+	if err != nil {
+		return nil, err
+	}
+
+	return starlark.String(loc.String()), nil
+}
+
+func sleep(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x starlark.Int
+	if err := starlark.UnpackArgs("sleep", args, kwargs, "x", &x); err != nil {
+		return starlark.None, err
+	}
+	dur, ok := x.Int64()
+	if !ok {
+		return starlark.None, fmt.Errorf("invalid sleep value")
+	}
+
+	time.Sleep(time.Duration(dur))
+	return starlark.None, nil
+}
+
+func parseTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x, location starlark.String
+		format      = starlark.String(time.RFC3339)
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location", &location); err != nil {
+		return nil, err
+	}
+
+	if location == "" {
+		t, err := time.Parse(format.String(), x.String())
+		if err != nil {
+			return nil, err
+		}
+		return Time(t), nil
+	}
+
+	loc, err := time.LoadLocation(location.String())
+	if err != nil {
+		return nil, err
+	}
+	t, err := time.ParseInLocation(format.String(), x.String(), loc)
+	if err != nil {
+		return nil, err
+	}
+	return Time(t), nil
+}
+
+func fromUnixTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x starlark.Int
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x); err != nil {
+		return nil, err
+	}
+
+	i, err := strconv.ParseInt(x.String(), 10, 64)
```

Comment:
x is already an int; there's no need to format and parse it.

var x int
if err := starlark.UnpackArgs("time", args, kwargs, "x", &x); err != nil { ... }
return Time(time.Unix(x, 0)), nil